src/core/annotation.js

```diff
@@ -1602,9 +1617,13 @@ class ButtonWidgetAnnotation extends WidgetAnnotation {
   }
 
   async _saveCheckbox(evaluator, task, annotationStorage) {
-    const defaultValue = this.data.fieldValue && this.data.fieldValue !== "Off";
     const value = annotationStorage[this.data.id];
+    if (value === undefined) {
+      // No change
```

Comment:
`undefined` actually means "not rendered" because otherwise the value would be false/true depending on the checked state. Given that we also don't have this comment in `WidgetAnnotation.save`, I'd say we can just remove this line here and in `_saveRadioButton` below.